### ISSUE TITLE
Unhelpful error when particle data is missing positions

### ISSUE THREAD

**Issue by mtrocadomoreira** [openPMD/openPMD-viewer#443]
Hello,

I was trying to read a dataset from a 2D WarpX simulation where the x and z positions for one of the species were not saved as a diagnostic. The "position" folder of the species in question in the HDF5 OpenPMD file therefore contains only an empty "y" folder.

Using `avail_record_components[species]` and passing those quantitites to `get_particle`, I get the following misleading error:

`KeyError: "Unable to synchronously open attribute (can't locate attribute: 'macroWeighted')"`

The exception is avoided if I manually remove "y" from the list of quantities. I am opening this issue because there is probably a better way to handle this.

Here is a minimal example of what I mean (I am not attaching the data file, though, because it's too large):

```python

from openpmd_viewer import OpenPMDTimeSeries

path = "path/to/file.h5"
species = "bunch"

op_obj = OpenPMDTimeSeries(path)

all_quants = op_obj.avail_record_components[species]

# The workaround:
# all_quants = list(set(all_quants) - {"y"})

all_data = op_obj.get_particle(all_quants, iteration=2760000, species=species)
```

I'm happy to provide more details, of course!

Thank you
